var/spack/repos/builtin/packages/vampirtrace/package.py

```diff
@@ -23,6 +23,12 @@ class Vampirtrace(AutotoolsPackage):
     depends_on('papi')
     depends_on('zlib')
 
+    # VampirTrace fails to build with newer versions of MPICH due to
+    # https://github.com/pmodels/mpich/commit/c3dbc09ae20a503ac4b870893e3e330d52ea5a3b
+    patch('mpi3-const.patch', when='^mpich@3.3:')
+    # VampirTrace fails to build with OpenMPI for the same reason
+    patch('mpi3-const.patch', when='^openmpi')
+
```

Comment:
@adamjstewart Is there a way to create a when clause that combines both of these cases?

Reply:
Nope, there is no "or" in the spec syntax.